setup.py

```diff
@@ -0,0 +1,204 @@
+from setuptools import setup
+
+import locale
+import os
+import re
+import subprocess
+import warnings
+
+from pathlib import Path
+from skyportal import __version__
+
+
+def _decode_stdio(stream):
+    try:
+        stdio_encoding = locale.getdefaultlocale()[1] or 'utf-8'
+    except ValueError:
+        stdio_encoding = 'utf-8'
+
+    try:
+        text = stream.decode(stdio_encoding)
+    except UnicodeDecodeError:
+        # Final fallback
+        text = stream.decode('latin1')
+
+    return text
+
+
+def get_git_devstr(sha=False, show_warning=True, path=None):
```

Comment:
This function is quite large, and seems like overkill.  It is only ever used with `sha=False`.  But, why?

Would the following suffice? https://github.com/skyportal/skyportal/blob/master/skyportal/__init__.py#L6